engine/gamesys/src/gamesys/components/comp_gui.cpp

```diff
@@ -1060,7 +1081,7 @@ namespace dmGameSystem
         dmGui::NodeType node_type = dmGui::GetNodeType(scene, first_node);
         assert(node_type == dmGui::NODE_TYPE_PARTICLEFX);
 
-        uint32_t vb_max_size = dmParticle::GetMaxVertexBufferSize(gui_world->m_ParticleContext, dmParticle::PARTICLE_GUI) - gui_world->m_RenderedParticlesSize;
+        uint32_t vb_max_size = dmParticle::GetMaxVertexBufferSize(gui_world->m_ParticleContext, sizeof(ParticleGuiVertex)) - gui_world->m_RenderedParticlesSize;
```

Comment:
> Not sure what the best way is to define the particle budget now.
> Previously it was bytes, but now that we can support multiple materials with different vertex formats, it might be relevant to think of #particles.
> 🤔

What budget are you referring to here? Both the GUI and the particle settings specify particle count (and components)

<img width="444" alt="image" src="https://github.com/defold/defold/assets/169640/fd3a7567-d10d-42c2-b8f0-b21ca41076d4">

> The usage of the sizeof(ParticleGuiVertex) seems a little strange now. Perhaps add a comment or two to show that the size is an estimate, and that the code is works ok?
> 
> E.g. the vertex count relies on that size

But for GUI the size isn't an estimate, it _is_ what is allocated regardless of what material is used 🤔 this is because we don't support any custom vertex formats in GUI right now, so we don't use the configuration from the material.